addons/account/models/mail_message.py

```diff
@@ -108,30 +130,57 @@ def _compute_account_audit_log_partner_id(self):
     def _search_account_audit_log_partner_id(self, operator, value):
         return self._search_audit_log_related_record_id('res.partner', operator, value)
 
+    def _get_extra_ids(self, model):
+        # Note: Under standard record rules, self.env['account.move.line'] is already limited
+        if model == 'res.partner':
+            return self.env['account.move.line'].sudo().search([
+                ('partner_id', '!=', False),
+                ('company_id.restrictive_audit_trail', '=', True),
+            ]).mapped('partner_id').ids
```

Comment:
This is not feasible on large databases

One could use sub-queries, for instance
```suggestion
            return self.env['account.move.line'].sudo()._search([
                ('partner_id', '!=', False),
                ('company_id.restrictive_audit_trail', '=', True),
            ]).select('partner_id')
```

This will return a `SQL` object that can be used in domains

Maybe we want `select('DISTINCT partner_id')` but I'm not sure, we should look at the query plan

Reply:
Thanks for the tip ! I hadn't in mind the differences between these options.
And I thing the DISTINCT keyword is not necessary, as we simply check if our id is in the result.